macOS/unwindmacos.cpp

```diff
@@ -414,8 +619,104 @@ size_t UnwindGeneratorMacOS::MaybeRedirectExecution(ModuleInfo* module, size_t I
   tinyinst_.CommitCode(module, code_size_before, (code_size_after - code_size_before));
 
   // we registered everything we have so far
-  unwind_data->have_data_to_register = false;
+  unwind_data->metadata_list.clear();
+  unwind_data->last_lookup = UnwindDataMacOS::LastLookup();
   
   // give the target process the address to continue from
   return continue_address;
 }
+
+size_t UnwindGeneratorMacOS::GetCustomPersonality(ModuleInfo* module, size_t original_personality) {
+  UnwindDataMacOS *unwind_data = (UnwindDataMacOS *)module->unwind_data;
+
+  auto iter = unwind_data->translated_personalities.find(original_personality);
+  if(iter != unwind_data->translated_personalities.end()) {
+    return iter->second;
+  }
+
+  size_t code_size_before = module->instrumented_code_allocated;
+
+  size_t new_personality_addr = tinyinst_.GetCurrentInstrumentedAddress(module);
+  
+  unsigned char assembly_part1[] = {
+    // function prologue
+    0x55, // push rbp
+    0x48, 0x89, 0xE5, // mov rbp, rsp
+    // save registers we're modifying
+    0x53,
```

Comment:
These four instructions are missing mnemonics in comments? :)

Reply:
This is entirely my fault as I wrote this assembly snippets. Anyway, the instructions here are 
```
53                      push   rbx
41 54                   push   r12
41 55                   push   r13
41 56                   push   r14
```